### PR TITLE
[processor/k8sattribute] run tests on 4 latest k8s version

### LINKED ISSUE #18514: [CICD] [e2e/k8s] Run tests on 4 latest kubernetes versions
We need to run the k8s e2e tests on all supported k8s versions, the latest 4: https://kubernetes.io/releases/

### PR DESCRIPTION
**Pull request by fatsheep9146**
**Description:** 
suopporun tests on 4 latest k8s version

**Link to tracking Issue:** <Issue number if applicable>
fix #18514
related #15651

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
LGTM

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -43,6 +46,8 @@ jobs:
         uses: helm/chart-testing-action@v2.3.0
       - name: Create kind cluster
         uses: helm/kind-action@v1.5.0
+        with:
+          kubectl_version: ${{ matrix.k8s-version }}
```
Comment (dmitryax) [inline .github/workflows/e2e-tests.yml:50]: Does it create k8s cluster with this version or just kubectl?

Comment (fatsheep9146) [reply .github/workflows/e2e-tests.yml:50]: My bad, this variable only effects the kubectl version, not cluster, I will check other variables.
@dmitryax

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -10,6 +10,9 @@ on:
 
 jobs:
   kubernetes-test:
+    strategy: 
+      matrix:
+        k8s-version: ["v1.26.0", "v1.25.3", "v1.24.7", "v1.23.13"]
```
Comment (dmitryax) [inline .github/workflows/e2e-tests.yml:15]: Do we need to specify the patch version or can be `v1.26` etc?

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (fatsheep9146) [reply .github/workflows/e2e-tests.yml:51]: Using `node_image` is right.
image

File: .github/workflows/e2e-tests.yml (hunk 2 above)
Comment (fatsheep9146) [reply .github/workflows/e2e-tests.yml:15]: https://github.com/kubernetes-sigs/kind/releases

I got the supported versions in this page, and did not see other version in 1.26 .

I also check the kind official docker hub, only `1.26.0` is available.

Comment (dmitryax) [reply .github/workflows/e2e-tests.yml:15]: Ok, sounds good. Thanks for confirming

**Review by dmitryax — APPROVED** [review]

**Pull request merged**